### DISCUSSION TITLE
Reprojection/Warping

### DISCUSSION THREAD

**Discussion by drwelby** [Toblerity/Fiona#1134] [Ideas]
`rasterio`'s `WarpedVRT` objects are lovely things - could Fiona behave similarly and have objects that emit reprojected data?

- maybe just `t_srs='EPSG:1234'` kwarg
- or a similar class to `WarpedVRT` that takes a Fiona reader

Is there some pitfall to letting GDAL do the reprojection work a la `ogr2ogr`?

**Comment by sgillies** [comment]
@drwelby I don't have much experience with OGR VRTs. A programmable abstraction for them could be handy.

One big difference between raster and vector datasets (and rasterio and fiona) is that raster windows (slices) are kinda arbitrary, while a vector dataset's records (thinking in terms of rows) are not. Reading a feature, and then reprojecting it, is the finest grained operation. OGR's VRT driver does exactly the same thing. I guess what I'm saying is that I'm not sure that

```python
with fiona.open("file.shp") as shp, fiona.VRT(shp, crs="EPSG:1234") as vrt:
    list(vrt)
```

is radically better than

```python
with fiona.open("file.shp") as shp:
    [transform_geom(shp.crs, "EPSG:1234", f["geometry"]) for f in shp]
```

I could be convinced otherwise!

**Reply by drwelby** [reply]
It's definitely not on the same magnitude of improvement compared to raster side but maybe I'll scratch that itch.

**Reply by sgillies** [reply]
Sticking the transformed geometry back onto the feature with its properties could be a win. If I may make a suggestion, a fiona VRT's state could easily be stored as an XML string, because OGR allows this

```python
with fiona.open("<VRT>...</VRT>") as vrt:
    ...
```
